deploy/lib/xquery/setup.xqy

```diff
@@ -2855,6 +2866,51 @@ declare function setup:validate-range-field-indexes($admin-config, $database, $d
       setup:validation-fail(fn:concat("Database mismatched range field index: ", $expected/db:field-name))
 };
 
+declare function setup:add-geospatial-region-path-indexes(
+  $admin-config as element(configuration),
+  $database as xs:unsignedLong,
+  $db-config as element(db:database)) as element(configuration)
+{
+  admin:database-add-geospatial-region-path-index(
+    setup:remove-existing-geospatial-region-path-indexes($admin-config, $database),
+    $database,
+    for $index in $db-config/db:geospatial-region-path-indexes/db:geospatial-region-path-index
+    return
+      if (setup:at-least-version("9.0-0")) then
+        admin:database-geospatial-region-path-index(
+          $index/db:path-expression,
+          $index/db:coordinate-system,
+          $index/db:geohash-precision,
+          ($index/db:invalid-values, $default-invalid-values)[1]
+        )
```

Comment:
I think this needs to be wrapped in xdmp:value or xdmp:eval, otherwise it will probably break on syntax check against ML 7 and 8..